nifi-extension-bundles/nifi-extension-utils/nifi-event-listen/src/main/java/org/apache/nifi/processor/util/listen/AbstractListenEventBatchingProcessor.java

```diff
@@ -45,27 +43,31 @@
  */
 public abstract class AbstractListenEventBatchingProcessor<E extends Event> extends AbstractListenEventProcessor<E> {
 
+    private static final List<PropertyDescriptor> PROPERTIES = List.of(
+            NETWORK_INTF_NAME,
+            PORT,
+            RECV_BUFFER_SIZE,
+            MAX_MESSAGE_QUEUE_SIZE,
+            MAX_SOCKET_BUFFER_SIZE,
+            CHARSET,
+            ListenerProperties.MAX_BATCH_SIZE,
+            ListenerProperties.MESSAGE_DELIMITER
+    );
+
     // it is only the array reference that is volatile - not the contents.
     protected volatile byte[] messageDemarcatorBytes;
 
     @Override
     protected void init(final ProcessorInitializationContext context) {
-        final List<PropertyDescriptor> descriptors = new ArrayList<>();
-        descriptors.add(NETWORK_INTF_NAME);
-        descriptors.add(PORT);
-        descriptors.add(RECV_BUFFER_SIZE);
-        descriptors.add(MAX_MESSAGE_QUEUE_SIZE);
-        descriptors.add(MAX_SOCKET_BUFFER_SIZE);
-        descriptors.add(CHARSET);
-        descriptors.add(ListenerProperties.MAX_BATCH_SIZE);
-        descriptors.add(ListenerProperties.MESSAGE_DELIMITER);
-        descriptors.addAll(getAdditionalProperties());
-        this.descriptors = Collections.unmodifiableList(descriptors);
-
-        final Set<Relationship> relationships = new HashSet<>();
-        relationships.add(REL_SUCCESS);
-        relationships.addAll(getAdditionalRelationships());
-        this.relationships = Collections.unmodifiableSet(relationships);
+        this.descriptors = Stream.concat(
+                PROPERTIES.stream(),
```

Comment:
Is there no simpler/cleaner option than Stream.concat ... list.stream().... .toList()?  Overall this is cleanear in some ways but does seem like there is a cleaner/simpler way of saying 'List from these lists'

Reply:
I used this as I saw that being used in other parts of the code e.g. `ElasticSearchLookupService`.

Reply:
It would surprise me if this is the best way to express these declarations.

Reply:
I thought it was the cleanest way to make an unmodifiable list.